### DISCUSSION TITLE
Pybricks on Linux TimeoutError when running simple project.

### DISCUSSION THREAD

**Discussion by bojan2501** [pybricks/support#1740] [Q&A] [answered]
Hi,
trying to run simple program on my hub:
```
from pybricks.hubs import InventorHub
from pybricks.parameters import Color
from pybricks.tools import wait

hub = InventorHub()
hub.light.on(Color.RED)
wait(5000)
```
with command:
`pybricksdev -d run ble main.py`
And getting:
`TimeoutError`

Checked is the device connected:
```
bluetoothctl devices Connected
Device 3B:D5:AE:99:02:02 Pybricks Hub
```
Then run with name param:
`pybricksdev -d run ble --name 3B:D5:AE:99:02:02 main.py`
Same error:
`TimeoutError`

Env: Linux(ArchLinux)
Lego Hub: InventorHub
Editor: PyCharm(not sure if relevant)

What could be a problem?

All best
Bojan

**Comment by dlech** [comment, marked as answer]
If the device is already connected when running `pybricksdev run`, then the device can't be found. This is because `pybricksedev` is expecting to scan an connect to the device. If a hub is connected, then it is not advertising and the scanner cannot see it because the scanner is looking for advertising data.

**Reply by bojan2501** [reply]
Hi @dlech,
Thank you for the answer. And that did the trick.
I was under impression that it needs to be paired and connected.
All the best
Bojan
